internal/storage/storage.go

```diff
@@ -39,16 +42,44 @@ type PackageVersion struct {
 	Name    string
 	Version string
 
+	root   string
```

Comment:
I didn't understand why this is needed for promotion.

Reply:
Same reason, missed during extraction. Fixed.